src/OSPSuite.Core/Serialization/Xml/OriginDataCacheXmlSerializer.cs

```diff
@@ -0,0 +1,64 @@
+using System.Xml.Linq;
+using OSPSuite.Core.Domain;
+using OSPSuite.Core.Domain.Builder;
+using OSPSuite.Utility.Collections;
+using OSPSuite.Utility.Extensions;
+
+namespace OSPSuite.Core.Serialization.Xml
+{
+   public class OriginDataCacheXmlSerializer : OSPSuiteXmlSerializer<OriginDataCache>
+   {
+      // public class CategoryItemXmlSerializer<TItem> : ObjectBaseXmlSerializer<TItem> where TItem : CategoryItem
+      // {
+      //    public override void PerformMapping()
+      //    {
+      //       base.PerformMapping();
+      //       Map(x => x.DisplayName);
+      //       Map(x => x.Category);
+      //    }
+      // }
+      //
+      // public class CalculationMethodXmlSerializer : CategoryItemXmlSerializer<CalculationMethod>
+      // {
+      //    public override void PerformMapping()
+      //    {
+      //       base.PerformMapping();
+      //       Map(x => x.AllModels);
+      //       Map(x => x.AllSpecies);
+      //    }
+      // }
+
+      protected OriginDataCacheXmlSerializer(string name)
+         : base(name)
+      {
+      }
+
+      protected OriginDataCacheXmlSerializer() 
+      {
+      }
+
+      public override void PerformMapping()
+      {
+         Map(x => x.ValueOrigin);
+      }
+
+      protected override void TypedDeserialize(OriginDataCache cache, XElement element, SerializationContext serializationContext)
+      {
+         base.TypedDeserialize(cache, element, serializationContext);
+         var cacheElement = element.Element(Constants.Serialization.VALUES);
+         var stringStringCacheSerializer = SerializerRepository.SerializerFor<Cache<string, string>>();
+         var values = stringStringCacheSerializer.Deserialize<Cache<string,string>>(cacheElement, serializationContext);
+         values.Keys.Each(x => cache.Add(x, values[x]));
+      }
+
+      protected override XElement TypedSerialize(OriginDataCache cache, SerializationContext serializationContext)
+      {
+         var element = base.TypedSerialize(cache, serializationContext);
```

Comment:
hum..Why do you have to do all of this? 
I think if the OridinDataCahce is derived from CAche (not sure what you gain here really), then use the Cahce serialzier
If your OriginData HAS a cache, serialize it 
The easiest would have been to dump everything as a string. Since we are creating objects, let's not use string and string but real objects?

OriginData has a _cache of OriginDataItem or whatever
OriginDataItem 
name (key)
displayName (==name if not set)
description = empty bydefault
icon? 

That way you can display your species with an icon, you can display a populationusing the display name coming from PKSim etc.

Either One string, or real structure. A Cache<String, string> (or an object derived from it), does not feel like a structure to use for an API exchange. THis is what I meant with the wrong abstraction and why I always advised against Cache in Public API (unless required as get for serialziation, which here also is more lazyness than anything)

Reply:
Updated with this. I know what you mean about the string dump being the simplest, but I would like to keep the separation of all the details into separate fields until the BB reaches MoBi for viewing.